src/activities/assignments/CategoriesEntity.js

```diff
@@ -110,16 +116,23 @@ export class CategoriesEntity extends Entity {
 		}
 	}
 
+	_hasCategoryIdChanged(categoryId) {
+		const selectedCategory = this.getSelectedCategory();
+		const initialId = selectedCategory && selectedCategory.categoryId;
+
+		return categoryId !== initialId;
+	}
+
 	async save(category) {
-		if (!this.canEditCategories()) return;
+		if (!this.canAddCategories()) return;
```

Comment:
Is this right? How do they select a new category when they don't have addCategory permissions then?